### PR TITLE
Feature/48042 Migrate rating - v2

### PR DESCRIPTION
**Pull request by jszczech**

### CODE REVIEW

File: components/src/molecules/rating/Rating.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    },
+    /**
+     * Icon for active element
+     */
+    activeIcon: {
+      type: String,
+      default: 'star'
+    },
+    /**
+     * Icon for not inactive element
```
Comment (Aekal) [inline components/src/molecules/rating/Rating.js]: Remove `not` work :D

File: components/src/molecules/rating/Rating.spec.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+  it('renders slot text when passed', () => {
+    const wrapper = mount(ARating, {
+      slots: {
+        item: `<span data-test="item">x</span>`
+      }
+    })
+
+    expect(wrapper.find('.a-rating > button > span').exists()).toBe(true)
+  })
```
Comment (Aekal) [inline components/src/molecules/rating/Rating.spec.js]: It'll be nice to have at least one test for checking the logic of rating, like clicking on 3rd star will return selected rate 3 or something like that - just to know if selecting on rating options works properly

File: components/src/molecules/rating/Rating.stories.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
+import { storiesOf } from '@storybook/vue'
+import { action } from '@storybook/addon-actions'
+
```
Comment (Aekal) [inline components/src/molecules/rating/Rating.stories.js]: Could you add knob for `items` prop to edit this number dynamically?
Maybe also color picker knobs for `active-color` and `unchecked-color`?

File: components/assets/images/rating/active-star.svg [hunk 4]
```diff
@@ -0,0 +1,3 @@
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 19">
+  <path d="M10 15.27L16.18 19l-1.64-7.03L20 7.24l-7.19-.61L10 0 7.19 6.63 0 7.24l5.46 4.73L3.82 19 10 15.27z" fill="#fab216"/>
+</svg>
```
[the remark this answers is not in the document]
Comment (jszczech) [reply components/assets/images/rating/active-star.svg]: Oh sorry, I forgot delete this file :D

File: components/src/molecules/rating/Rating.js (hunk 1 above)
Comment (jszczech) [reply components/src/molecules/rating/Rating.js]: :v

File: components/src/molecules/rating/Rating.stories.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+      <a-rating
+        :items="3"
+      >
+        <template #item="data">
+          <button
+            style="background: none;"
+            @click="click(data.item)"
+          >
+            {{ data.item.val }}
+          </button>
```
Comment (szafran89) [inline components/src/molecules/rating/Rating.stories.js]: it's a little weird b/c it generates button inside button, please add an example with svg icon added inline.

File: components/src/molecules/rating/Rating.spec.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+  it('renders slot text when passed', () => {
+    const wrapper = mount(ARating, {
+      slots: {
+        item: `<span data-test="item">x</span>`
+      }
+    })
+
+    expect(wrapper.find('.a-rating > button > span').exists()).toBe(true)
```
Comment (szafran89) [inline components/src/molecules/rating/Rating.spec.js]: hmm, you don't check slot text but only if span tag exists, use `text` or `html` method

File: components/src/molecules/rating/Rating.stories.js [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+        :items="numberOfRatingKnob"
+        :active-icon="activeIcon"
+        :unchecked-icon="uncheckedIcon"
+        :active-color="activeColor"
+        :uncheckedColor="uncheckedColor"
+        @select="select"
+      />
+    `
+  }))
+  .add('Read only', () => ({
```
Comment (szafran89) [inline components/src/molecules/rating/Rating.stories.js]: can you add knobs for this story?

File: components/src/molecules/rating/Rating.stories.js [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+        default: number('Number of rating', 5)
+      }
+    },
+    template: `
+      <a-rating
+        :items="numberOfRatingKnob"
+        :active-icon="activeIcon"
+        :unchecked-icon="uncheckedIcon"
+        :active-color="activeColor"
+        :uncheckedColor="uncheckedColor"
```
Comment (szafran89) [inline components/src/molecules/rating/Rating.stories.js]: `: unchecked-color `

File: components/src/molecules/rating/Rating.scss [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+  &__button {
+    padding: $rating__button-padding;
+  }
+
+  &__icon {
+    width: $rating__icon-width;
+    height: $rating__icon-height;
+
+    &--disabled {
+      position: absolute;
```
[the remark this answers is not in the document]
Comment (jszczech) [reply components/src/molecules/rating/Rating.scss]: Yes, it's only for a partial star which should be on the top

File: components/src/molecules/rating/Rating.js [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+     * Class for button element
+     */
+    buttonClass: {
+      type: String,
+      default: null
+    }
+  },
+  data () {
+    return {
+      localItems: Array.from(Array(this.items), (x, index) => index + 1)
```
[the remark this answers is not in the document]
Comment (szafran89) [reply components/src/molecules/rating/Rating.js]: what about this?

[the remark this answers is not in the document]
Comment (jszczech) [reply components/src/molecules/rating/Rating.js]: Sorry, I missed that. In this case, I could move this operation to mounted ;)

**Pull request merged**